### PR TITLE
Occlusion Edge Blur 

### PR DESCRIPTION
**Pull request by rarattay**
Blurring the rough edges due to occluding using raw depth or body mask data 
- OcculsionMask shader combines both the depth and body index map into one mask that determines what is occluded by the real world
- Blur shader blurs texture assigned to it by using an average of the pixels around it 
-HoloAlpha reads from Occlusion shader to assign alpha values to the hologram

### CODE REVIEW

File: src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectFrameProvider.cpp [hunk 1]
```diff
@@ -210,25 +215,41 @@ uint8_t* AzureKinectFrameProvider::GetBodyIndexBuffer(k4a_capture_t capture)
...
+
+        bufferIndex++;
+    }
+}
+
+void AzureKinectFrameProvider::SetTransformedBodyMaskBuffer(uint16_t* transformedBodyMaskBuffer, int bufferSize)
+{
+    int bufferIndex = 0;
+
+    while (bufferIndex < bufferSize)
```
Comment (matthejo) [inline src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectFrameProvider.cpp]: >while (bufferIndex < bufferSize) [](start = 4, length = 32)

I would find a for loop more intuitive to read here than a while loop with an increment at the end

File: src/SpectatorView.Unity/Assets/SpectatorView.Editor/Shaders/Blur.shader [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            Texture2D _MaskTexture;
+            float _BlurIntensity;
+            SamplerState sampler_point_clamp;
+
+            fixed4 frag(v2f i) : SV_Target
+            {
+                half4 blurVal = fixed4(0,0,0,0);
+                float sumX = 0.0;
+
+                sumX += _MaskTexture.Sample(sampler_point_clamp, float2(i.uv[0] - 5.0 * _BlurIntensity, i.uv[1])).r * 0.025;
```
Comment (matthejo) [inline src/SpectatorView.Unity/Assets/SpectatorView.Editor/Shaders/Blur.shader:51]: >0.025 [](start = 118, length = 5)

If these magic numbers mean anything or came from somewhere, can we reference that in a comment?

**Review by matthejo — APPROVED** [review]
:shipit:

File: src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectFrameProvider.cpp (hunk 1 above)
Comment (rarattay) [reply src/SpectatorView.Native/SpectatorView.Compositor/Compositor/AzureKinectFrameProvider.cpp]: Agreed

File: src/SpectatorView.Unity/Assets/SpectatorView.Editor/Shaders/Blur.shader (hunk 2 above)
Comment (rarattay) [reply src/SpectatorView.Unity/Assets/SpectatorView.Editor/Shaders/Blur.shader:51]: Added comment

**Pull request merged**